src/expr/src/scalar/func/impls/time.rs

```diff
@@ -28,11 +28,12 @@ sqlfunc!(
     #[sqlname = "timetoiv"]
     #[preserves_uniqueness = true]
     fn cast_time_to_interval<'a>(t: NaiveTime) -> Result<Interval, EvalError> {
-        Interval::new(
-            0,
-            t.num_seconds_from_midnight() as i64,
-            t.nanosecond().into(),
-        )
-        .map_err(|_| EvalError::IntervalOutOfRange)
+        // wont overflow because value can't exceed 24 hrs + 1_000_000 ns = 86_400 seconds + 1_000_000 ns = 86_400_001_000 us
+        let micros: i64 = i64::from(t.num_seconds_from_midnight())
```

Comment:
Could this use `convert_date_time_unit`?

Reply:
I changed it to use `convert_date_time_unit` for the second to microsecond conversion, but that function doesn't support nanosecond so that stayed manual.